static/src/js/components/AccessMethod/AccessMethod.jsx

```diff
@@ -57,260 +41,105 @@ const EchoForm = lazy(() => import('./EchoForm'))
-    this.handleToggleSpatialSubsetting = this.handleToggleSpatialSubsetting.bind(this)
-    this.handleConcatenationSelection = this.handleConcatenationSelection.bind(this)
-    this.handleHarmonySelection = this.handleHarmonySelection.bind(this)
-    this.handleSwoldrOptions = this.handleSwoldrOptions.bind(this)
-    this.handleCollectionGranuleListUpdate = this.handleCollectionGranuleListUpdate.bind(this)
-    this.handleRasterResolutionUpdate = this.handleRasterResolutionUpdate.bind(this)
-    this.handleSampleGrid = this.handleSampleGrid.bind(this)
-    this.handleGranuleExtent = this.handleGranuleExtent.bind(this)
+const AccessMethod = ({
+  accessMethods,
+  index,
+  isActive,
+  metadata,
+  onSelectAccessMethod,
+  onSetActivePanel,
+  onTogglePanels,
+  onUpdateAccessMethod,
+  selectedAccessMethod,
+  shapefileId,
+  spatial,
+  temporal,
+  ursProfile,
+  granuleMetadata,
+  projectCollection
+}) => {
+  const { [selectedAccessMethod]: selectedMethod = {} } = accessMethods
+
+  const {
+    form,
+    rawModel = null,
+    selectedVariables = [],
+    selectedOutputFormat,
+    selectedOutputProjection,
+    supportedOutputFormats = [],
+    supportedOutputProjections = [],
+    supportsTemporalSubsetting = false,
+    supportsShapefileSubsetting = false,
+    supportsBoundingBoxSubsetting = false,
+    supportsVariableSubsetting = false,
+    supportsConcatenation = false,
+    supportsSwodlr = false,
+    defaultConcatenation = false,
+    enableTemporalSubsetting: hasEnabledTemporalSubsetting
+  } = selectedMethod || {}
+
+  const { isRecurring } = temporal
+
+  // EnabledTemporalSubsetting by default
+  let setTemporal = true
+
+  // If enabledTemporalSubsetting is explictly false, set the initial value to false
+  if (hasEnabledTemporalSubsetting === false) {
+    setTemporal = false
   }
 
-  UNSAFE_componentWillReceiveProps() {
-    const { temporal, granuleMetadata, projectCollection } = this.props
-
-    const { isRecurring } = temporal
-
-    // Disable temporal subsetting if the user has a recurring date selected
-    if (isRecurring) {
-      this.setState({
-        enableTemporalSubsetting: false
-      })
-    }
-
-    const {
-      granules: projectCollectionGranules = {}
-    } = projectCollection
-
-    const {
-      addedGranuleIds = [],
-      allIds: granulesAllIds = []
-    } = projectCollectionGranules
-
-    let granulesToDisplay = []
+  // Initialize State Variables
+  const [enableTemporalSubsetting, setEnableTemporalSubsetting] = useState(setTemporal)
+  const [selectedHarmonyMethodName, setSelectedHarmonyMethodName] = useState('')
+  const [isHarmony, setIsHarmony] = useState(selectedAccessMethod.startsWith('harmony'))
+  const [enableSpatialSubsetting, setEnableSpatialSubsetting] = useState(false)
+  const [enableConcatenateDownload, setEnableConcatenateDownload] = useState(defaultConcatenation)
+  const [granuleList, setGranuleList] = useState([])
+  // Only set state on re-render
```

Comment:
Does comment belong? I'm not sure what it is referencing here

Reply:
Removed.